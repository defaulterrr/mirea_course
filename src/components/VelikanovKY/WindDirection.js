const WindDirection = ({ deg }) => {
  let direction = 'No way';
  if (deg >= 349 || deg < 12) { direction = 'North'; } else if (deg >= 12 && deg < 34) { direction = 'North-northeast'; } else if (deg >= 34 && deg < 57) { direction = 'Northeast'; } else if (deg >= 57 && deg < 80) { direction = 'East-northeast'; } else if (deg >= 80 && deg < 102) { direction = 'East'; } else if (deg >= 102 && deg < 125) { direction = 'East-southeast'; } else if (deg >= 125 && deg < 147) { direction = 'Southeast'; } else if (deg >= 147 && deg < 170) { direction = 'South-southeast'; } else if (deg >= 170 && deg < 192) { direction = 'South'; } else if (deg >= 192 && deg < 214) { direction = 'South-southwest'; } else if (deg >= 214 && deg < 237) { direction = 'Southwest'; } else if (deg >= 237 && deg < 260) { direction = 'West-southwest'; } else if (deg >= 260 && deg < 282) { direction = 'West'; } else if (deg >= 282 && deg < 304) { direction = 'West-northwest'; } else if (deg >= 304 && deg < 326) { direction = 'Northwest'; } else if (deg >= 326 && deg < 349) { direction = 'Nortn-northwest'; }
  return (direction);
};
export default WindDirection;
