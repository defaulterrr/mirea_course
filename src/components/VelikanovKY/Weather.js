import React, {useEffect, useState} from 'react';
import WindDirection from './WindDirection';
const APP_ID = "719dddbed313225204fd616e1b75e83f";
const getUpdateUrl = (city, countryCode = '') => `https://api.openweathermap.org/data/2.5/weather?q=${city},${countryCode}&appid=${APP_ID}`;
const fetchData = async (weatherUpdate) => {
   try {
    const response = await fetch(getUpdateUrl('Moscow','ru')) 
    const json = await response.json();
    console.log(json);
        weatherUpdate({
            temperature: json.main.temp,
            city: json.name,
            icon: 'https://openweathermap.org/img/wn/' + json.weather[0].icon +'@2x.png',
            feel: json.main.feels_like,
            description: json.weather[0].description,
            wind: json.wind.speed,
            deg: json.wind.deg,
            isLoading: false
        }) 
   } catch (error) {
       console.log("Cathed error: " + error);
   }
    
    
    
}

const Weather = () => {
    const STYLE = {
        fontSize: '10pt',
    }
    const DIV_STYLE = {
        border:                 '1px solid black',
        borderRadius:                      '25px',
        backgroundColor:  'rgba(255,255,255,0.1)',
        padding:                            '5px',
    }
    const [ weather,setWeather ] = useState({
        city:  'Moscow,ru',
        isLoading:  true    
    })
    useEffect(() => {
        fetchData(setWeather)
    },[])
    
    if (!weather.isLoading){
        return (<div style = {DIV_STYLE}>
            Weather in {weather.city}
            <div>
                <img className="weather-widget__img" src={weather.icon} alt="Weather in Moscow, RU" width="50" height="50" />
                {parseInt(weather.temperature -273.15)}°C <br />
                {weather.description}, feels like {parseInt(weather.feel - 273.15)}°C <br />
                <div style = {STYLE}>{weather.wind} m/s <br /> <WindDirection deg = {weather.deg} /> ({weather.deg})</div>
            </div>
        </div>
        )
    }
    else return <h3>Weather in {weather.city}</h3>
}

export default Weather;