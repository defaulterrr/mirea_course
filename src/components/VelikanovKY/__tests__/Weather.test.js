import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import Weather from '../Weather';
import WindDirection from '../WindDirection';
import { render } from '@testing-library/react';

const testData = {
  coord: { lon: 139, lat: 35 },
  weather: [
    {
      id: 800,
      main: 'Clear',
      description: 'clear sky',
      icon: '01n',
    },
  ],
  base: 'stations',
  main: {
    temp: 281.52,
    feels_like: 278.99,
    temp_min: 280.15,
    temp_max: 283.71,
    pressure: 1016,
    humidity: 93,
  },
  wind: {
    speed: 0.47,
    deg: 107.538,
  },
  clouds: {
    all: 2,
  },
  dt: 1560350192,
  sys: {
    type: 3,
    id: 2019346,
    message: 0.0065,
    country: 'JP',
    sunrise: 1560281377,
    sunset: 1560333478,
  },
  timezone: 32400,
  id: 1851632,
  name: 'London,uk',
  cod: 200,
};

let container = null;
beforeEach(() => {
  // подготавливаем DOM-элемент, куда будем рендерить
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // подчищаем после завершения
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('Should be a word indicating the loading', () => {
  const { getByText } = render(<Weather />);
  const linkElement = getByText("Fetching...");
  expect(linkElement).toBeInTheDocument();
});

test('WeatherDirection Snapshot', () => {
  const direction = renderer.create(<WindDirection deg={testData.wind.deg} />);
  expect(direction.toJSON()).toMatchSnapshot();
});
