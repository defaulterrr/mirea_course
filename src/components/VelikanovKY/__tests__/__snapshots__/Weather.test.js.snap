// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDirection Snapshot 1`] = `"East-southeast"`;
